### PR TITLE
Disable Calls to Identity Pallet

### PR DESCRIPTION
**Pull request by joepetrowski**
This PR filters calls from the Identity pallet from all Relay Chain runtimes as preparation to move the identity state and logic to a system parachain within each network.

After this change is deployed to a runtime, no more changes such as adding new sub-identities will be possible. The frozen state will be part of the genesis state of the system chain. After the system chain launches, the pallet and all state will be removed from each Relay Chain.

Applications and UIs that render display information from this pallet will need to read from the system chain when it launches.

### CODE REVIEW

File: polkadot/runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -157,11 +158,16 @@ pub fn native_version() -> NativeVersion {
...
-	fn contains(_c: &RuntimeCall) -> bool {
-		true
+/// Disable all calls to the Identity pallet. This will lock the state of the pallet, preventing
+/// further updates to identities and sub-identities. The locked state will be the genesis state of
+/// a new system chain and then removed from the Relay Chain.
+pub struct IdentityCalls;
+impl Contains<RuntimeCall> for IdentityCalls {
+	fn contains(c: &RuntimeCall) -> bool {
+		match c {
+			RuntimeCall::Identity(_) => true,
```
Comment (KiChjang) [inline polkadot/runtime/kusama/src/lib.rs:168]: This looks a bit weird, as the comment states that it's disabling all calls to the identity pallet, yet here it's returning true only for calls to the identity pallet.

File: polkadot/runtime/rococo/src/lib.rs [hunk 2]
```diff
@@ -134,11 +135,16 @@ pub fn native_version() -> NativeVersion {
...
+/// Disable all calls to the Identity pallet. This will lock the state of the pallet, preventing
+/// further updates to identities and sub-identities. The locked state will be the genesis state of
+/// a new system chain and then removed from the Relay Chain.
+pub struct IdentityCalls;
+impl Contains<RuntimeCall> for IdentityCalls {
+	fn contains(c: &RuntimeCall) -> bool {
+		match c {
+			RuntimeCall::Identity(_) => true,
+			_ => false,
+		}
```
Comment (ggwpez) [inline polkadot/runtime/rococo/src/lib.rs:147]:
```suggestion
		matches!(c, RuntimeCall::Identity(_))
```

**Review by ggwpez — APPROVED** [review]
There is a similar MR over here that is looking for a second review 🙏  https://github.com/paritytech/polkadot-sdk/pull/1303

**Comment by gilescope** [comment]
Will this prevent identity calls within a `utility.batch` or a `utility.asDerivative` call ? (Ideally we should have a generic base call filter that can walk containers)

**Comment by joepetrowski** [comment]
> Will this prevent identity calls within a `utility.batch` or a `utility.asDerivative` call ? (Ideally we should have a generic base call filter that can walk containers)

Yes, it does

**Comment by bkchr** [comment]
This pr should be opened against the fellowship runtimes repo.

**Comment by joepetrowski** [comment]
https://github.com/polkadot-fellows/runtimes/pull/25

**Review by muharem — APPROVED** [review]
was curious what an error a user will see, its going to be `CallFiltered` - [link](https://github.com/paritytech/polkadot-sdk/blob/7a3aefcaa842a3a1d92631b1ffc03d9503272ab7/substrate/frame/system/src/lib.rs#L560)

**Review by KiChjang — APPROVED** [review]
Best get this merged ASAP before we have to redo this again in the fellowship repo.

**Comment by joepetrowski** [comment]
> Best get this merged ASAP before we have to redo this again in the fellowship repo.

We will still need it in both repos AFAIK because Rococo/Westend will stay here but Polkadot/Kusama will only be synced with the 1.1.0 commit.

**Comment by xlc** [comment]
@joepetrowski could you write something at https://forum.polkadot.network about this? People need to know this feature will be paused for some time. How long do you expect for the down time? And combined with https://github.com/polkadot-fellows/runtimes/pull/31, I want to see a written down plan.

**Comment by xlc** [comment]
And this is going to conflict with https://github.com/paritytech/polkadot-sdk/pull/1291 a bit. We need to make sure the fellowship can be a registrar on the new chain as well.

**Comment by joepetrowski** [comment]
> And this is going to conflict with #1291 a bit. We need to make sure the fellowship can be a registrar on the new chain as well.

That is not a conflict, it's just a feature request for the new chain.

**Comment by xlc** [comment]
I am not saying this is going to cause problem, but more work will be required. At least my old testing script have to be updated.

**Comment by joepetrowski** [comment]
> @joepetrowski could you write something at https://forum.polkadot.network about this? People need to know this feature will be paused for some time. How long do you expect for the down time? And combined with [polkadot-fellows/runtimes#31](https://github.com/polkadot-fellows/runtimes/pull/31), I want to see a written down plan.

The PR description is a written plan. Yes I am working on something to share on the forum and in a Fellowship RFC. It should not be too shocking, the goal of system parachains has always been to migrate functionality off the Relay Chain (mentioned way back in [2019](https://github.com/paritytech/polkadot/issues/323)).

**Comment by joepetrowski** [comment]
> I am not saying this is going to cause problem, but more work will be required. At least my old testing script have to be updated. In order words, the fellowship as registrar work have to be paused until the new chain is up.

Yes, Polkadot is upgradeable and sometimes testing scripts need to be updated. Do you know that the Fellowship wants to act as a registrar? And if so, having Identity on a parachain seems like a perfect way to eat our own dog food and show a cross-parachain application.

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-2-0/4451/1

**Pull request merged**